resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/submit/submitsteps/initcontainer/SubmittedResourcesInitContainerStepSuite.scala

```diff
@@ -0,0 +1,173 @@
+    assert(returnedInitContainer.initContainer.getName === "mountedSecret")
+    assert(returnedInitContainer.podToInitialize.getMetadata.getLabels.asScala
+      === Map("mountedSecret" -> "true"))
+    assert(returnedInitContainer.initContainerDependentResources.length == 1)
+    val secret = returnedInitContainer.initContainerDependentResources.head.asInstanceOf[Secret]
+    assert(secret.getData === RSS_SECRET)
+    assert(secret.getMetadata.getName == RESOURCE_SECRET_NAME)
+    val expectedinitContainerProperties = Map(
+      RESOURCE_STAGING_SERVER_URI.key -> STAGING_SERVER_URI,
+      INIT_CONTAINER_DOWNLOAD_JARS_RESOURCE_IDENTIFIER.key -> JARS_RESOURCE_ID,
+      INIT_CONTAINER_DOWNLOAD_JARS_SECRET_LOCATION.key ->
+        s"$SECRET_MOUNT_PATH/$INIT_CONTAINER_SUBMITTED_JARS_SECRET_KEY",
+      INIT_CONTAINER_DOWNLOAD_FILES_RESOURCE_IDENTIFIER.key -> FILES_RESOURCE_ID,
+      INIT_CONTAINER_DOWNLOAD_FILES_SECRET_LOCATION.key ->
+        s"$SECRET_MOUNT_PATH/$INIT_CONTAINER_SUBMITTED_FILES_SECRET_KEY",
+      RESOURCE_STAGING_SERVER_SSL_ENABLED.key -> false.toString)
+    assert(returnedInitContainer.initContainerProperties === expectedinitContainerProperties)
+    assert(returnedInitContainer.additionalDriverSparkConf ===
+      Map(
+        EXECUTOR_INIT_CONTAINER_SECRET.key -> RESOURCE_SECRET_NAME,
+        EXECUTOR_INIT_CONTAINER_SECRET_MOUNT_DIR.key -> SECRET_MOUNT_PATH))
+  }
+
+  test ("testing prepareInitContainer w/ CERT and TrustStore Files w/o SSL") {
+    val submittedResourceStep = new SubmittedResourcesInitContainerStep(
+      RESOURCE_SECRET_NAME,
+      STAGING_SERVER_URI,
+      SECRET_MOUNT_PATH,
+      false,
+      TRUSTSTORE_URI,
+      CERT_URI,
+      Some(TRUSTSTORE_PASS),
+      Some(TRUSTSTORE_TYPE),
+      submittedDependencyUploader,
+      submittedResourcesSecretPlugin
+    )
+    val returnedInitContainer =
+      submittedResourceStep.prepareInitContainer(InitContainerSpec(
+        Map.empty[String, String],
+        Map.empty[String, String],
+        new Container(),
+        new Container(),
+        new Pod(),
+        Seq.empty[HasMetadata]))
+    val expectedinitContainerProperties = Map(
+      RESOURCE_STAGING_SERVER_URI.key -> STAGING_SERVER_URI,
+      INIT_CONTAINER_DOWNLOAD_JARS_RESOURCE_IDENTIFIER.key -> JARS_RESOURCE_ID,
+      INIT_CONTAINER_DOWNLOAD_JARS_SECRET_LOCATION.key ->
+        s"$SECRET_MOUNT_PATH/$INIT_CONTAINER_SUBMITTED_JARS_SECRET_KEY",
+      INIT_CONTAINER_DOWNLOAD_FILES_RESOURCE_IDENTIFIER.key -> FILES_RESOURCE_ID,
+      INIT_CONTAINER_DOWNLOAD_FILES_SECRET_LOCATION.key ->
+        s"$SECRET_MOUNT_PATH/$INIT_CONTAINER_SUBMITTED_FILES_SECRET_KEY",
+      RESOURCE_STAGING_SERVER_SSL_ENABLED.key -> false.toString) ++
+    Map(
+      RESOURCE_STAGING_SERVER_TRUSTSTORE_PASSWORD.key -> TRUSTSTORE_PASS,
+      RESOURCE_STAGING_SERVER_TRUSTSTORE_TYPE.key -> TRUSTSTORE_TYPE,
+      RESOURCE_STAGING_SERVER_TRUSTSTORE_FILE.key -> (SECRET_MOUNT_PATH + "/trustStore"),
```

Comment:
The logic here mounts the user's secret files as a secret volume into the pod. When you create a secret, you provide a map of key-value pairs. One of these keys for the trustStore is `trustStore` (see `constants.scala`). When secrets are mounted onto the disk, one provides a base path and a file is created for every secret data item in the secret, with the file name being the secret's key. Since we mount the secret at `SECRET_MOUNT_PATH` and the secret key is `trustStore`, the resolved path is `SECRET_MOUNT_PATH/trustStore`.

To make this clearer, use the constant from `constants.scala` in the unit test.